ImageViewer/Source/ImageFadeInHandler.swift

```diff
@@ -0,0 +1,24 @@
+//
+//  ImageFadeInHandler.swift
+//  ImageViewer
+//
+//  Created by Kristian Angyal on 02/03/2016.
+//  Copyright © 2016 MailOnline. All rights reserved.
+//
+
+import Foundation
+
+final class ImageFadeInHandler {
+    
+    var presentedImages:[Int] = []
+    
+    func imagePresentedAtIndex(index: Int) {
+        
```

Comment:
The naming here is a bit weird, it seems you are going to return something, but actually you are adding something. Maybe: `addPresentedImageIndex`. The "At" is also confusing, because it seems you care about the Index internally, but you really don't: `presentedImages.append(index)`. You just want to append into an array.